ui/components/ui/detailed-error-message/detailed-error-message.component.js

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const DetailedErrorMessage = (props, context) => {
+  const { errorMessage, errorKey, onErrorMessageClick, linkText } = props;
+  const error = errorKey ? context.t(errorKey) : errorMessage;
+
+  return (
+    <div className="error-message">
+      <img
+        src="./images/alert-red.svg"
+        alt=""
+        className="error-message__icon"
+      />
+      <div className="error-message__text">
+        {error}
+        {linkText ? (
+          <a className="error-message__link" onClick={onErrorMessageClick}>
+            {` ${linkText}`}
+          </a>
+        ) : (
+          ''
+        )}
+      </div>
+    </div>
+  );
+};
+
+DetailedErrorMessage.propTypes = {
+  errorMessage: PropTypes.string,
+  errorKey: PropTypes.string,
+  linkText: PropTypes.string,
+  onErrorMessageClick: PropTypes.func,
```

Comment:
Could we make these mandatory? I can't imagine we'd have a "detailed error message" that had no way to click to see the details.
```suggestion
  linkText: PropTypes.string.isRequired,
  onErrorMessageClick: PropTypes.func.isRequired,
```

This would let us simplify the render function as well, by removing the conditions that check if these props are present or not. And it would help validate that `linkText` is always set with `onErrorMessageClick`.